tests/trinity/integration/test_lightchain_integration.py

```diff
@@ -33,31 +42,121 @@
 )
 
 
+@pytest.fixture(scope='module')
+def geth_port():
+    sock = socket.socket()
+    sock.bind(('127.0.0.1', 0))
+    port = sock.getsockname()[1]
+    sock.close()
+    return port
```

Comment:
I found this the other day which might be of interest:

https://github.com/pytest-dev/pytest-asyncio#unused_tcp_port

The `pytest-asyncio` library has a pytest fixture for doing this.